src/main/java/org/usfirst/frc4904/robot/VisionTargetHub.java

```diff
@@ -0,0 +1,104 @@
+
+    public enum VisionTargetType {
+        HIGH_PORT(0), LOADING_PORT(1);
+
+        public int targetType;
+
+        private VisionTargetType(int targetType) {
+            this.targetType = targetType;
+        }
+    }
+
+    public VisionTargetHub(NetworkTableEntry targetTypesEntry, NetworkTableEntry distancesEntry,
+            NetworkTableEntry betasEntry, NetworkTableEntry thetasEntry) {
+        VisionTargetHub.targetTypesEntry = targetTypesEntry;
+        VisionTargetHub.distancesEntry = targetTypesEntry;
+        VisionTargetHub.betasEntry = targetTypesEntry;
+        VisionTargetHub.thetasEntry = targetTypesEntry;
+
+        enableDataCollection();
+        getTargetData();
+        createTargets();
+    }
+
+    @Override
+    public void periodic() {
+        if (isEnabled) {
+            getTargetData();
+            createTargets();
+        }
+    }
+
+    public void enableDataCollection() {
+        isEnabled = true;
+    }
+
+    public void disableDataCollection() {
+        isEnabled = false;
+    }
+
+    public void getTargetData() {
+        VisionTargetHub.targetTypes = targetTypesEntry.getDoubleArray(DEFAULT_INPUT_ARRAY);
+        VisionTargetHub.distances = distancesEntry.getDoubleArray(DEFAULT_INPUT_ARRAY);
+        VisionTargetHub.betas = betasEntry.getDoubleArray(DEFAULT_INPUT_ARRAY);
+        VisionTargetHub.thetas = thetasEntry.getDoubleArray(DEFAULT_INPUT_ARRAY);
+    }
+
+    public VisionTarget[] createTargets() {
+        VisionTargetHub.targets = new VisionTarget[targetTypes.length];
+        for (int i = 0; i < VisionTargetHub.targets.length; i++) {
+            VisionTargetHub.targets[i] = new VisionTarget(targetTypes[i], distances[i], betas[i], thetas[i]);
+        }
+        return VisionTargetHub.targets;
+    }
+
+    public VisionTarget[] filterTargetsByTargetType(VisionTarget[] targets, VisionTargetType intendedTargetType) {
+        return (VisionTarget[]) Arrays.stream(targets).filter(target -> target.checkTargetType(intendedTargetType))
+                .toArray();
+    }
+
+    public VisionTarget[] filterTargetsByTargetType(VisionTargetType intendedTargetType) {
+        return filterTargetsByTargetType(VisionTargetHub.targets, intendedTargetType);
+    }
+
+    public VisionTarget findClosestTarget(VisionTarget[] targets) {
+        return Arrays.stream(targets).min((target1, target2) -> Double.compare(target1.distance, target2.distance))
+                .get();
+    }
+
+    public VisionTarget findClosestTarget() {
+        return findClosestTarget(VisionTargetHub.targets);
+    }
+
+    public VisionTarget findIdealTarget(VisionTarget[] targets, VisionTargetType intendedTargetType) {
+        VisionTarget[] usableTargets = filterTargetsByTargetType(intendedTargetType);
+        return findClosestTarget(usableTargets);
+    }
+
+    public VisionTarget findIdealTarget(VisionTargetType intendedTargetType) {
+        return findIdealTarget(VisionTargetHub.targets, intendedTargetType);
```

Comment:
Doesn't really matter but the nice way to do this is have the filter and find closest methods work on streams, then you only iterate over it once